iotdb-2.0/src/main/java/cn/edu/tsinghua/iot/benchmark/iotdb200/DMLStrategy/TableSessionManager.java

```diff
@@ -0,0 +1,175 @@
+import org.apache.tsfile.file.metadata.enums.TSEncoding;
+import org.apache.tsfile.write.record.Tablet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class TableSessionManager extends SessionManager {
+  private static final Logger LOGGER = LoggerFactory.getLogger(TableSessionManager.class);
+  private final ITableSession tableSession;
+  private boolean isFirstExecution = true;
+
+  public TableSessionManager(DBConfig dbConfig) throws IoTDBConnectionException {
+    super(dbConfig);
+    List<String> hostUrls = new ArrayList<>(dbConfig.getHOST().size());
+    for (int i = 0; i < dbConfig.getHOST().size(); i++) {
+      hostUrls.add(dbConfig.getHOST().get(i) + ":" + dbConfig.getPORT().get(i));
+    }
+    this.tableSession = builderTableSession(hostUrls);
+  }
+
+  @Override
+  public void executeNonQueryStatement(String sql)
+      throws IoTDBConnectionException, StatementExecutionException {
+    tableSession.executeNonQueryStatement(sql);
+  }
+
+  @Override
+  public SessionDataSet executeQueryStatement(String sql)
+      throws IoTDBConnectionException, StatementExecutionException {
+    return tableSession.executeQueryStatement(sql);
+  }
+
+  @Override
+  public SessionDataSet executeQueryStatement(String sql, long timeoutInMs)
+      throws IoTDBConnectionException, StatementExecutionException {
+    return tableSession.executeQueryStatement(sql, timeoutInMs);
+  }
+
+  @Override
+  protected void insertRecord(
+      String deviceId,
+      long time,
+      List<String> measurements,
+      List<TSDataType> types,
+      List<Object> values) {
+    // nothing to do
+  }
+
+  @Override
+  protected void insertRecords(
+      List<String> deviceIds,
+      List<Long> times,
+      List<List<String>> measurementsList,
+      List<List<TSDataType>> typesList,
+      List<List<Object>> valuesList) {
+    // nothing to do
+  }
+
+  @Override
+  public void insertTablet(Tablet tablet, DeviceSchema deviceSchema)
+      throws IoTDBConnectionException, StatementExecutionException {
+    if (isFirstExecution) {
+      StringBuilder sql = new StringBuilder();
+      sql.append("use ").append(dbConfig.getDB_NAME()).append("_").append(deviceSchema.getGroup());
+      tableSession.executeNonQueryStatement(sql.toString());
+      isFirstExecution = false;
+    }
+    tableSession.insert(tablet);
+  }
+
+  @Override
+  public void init() {
+    // TableSession combines the build and open operations of the session
+  }
+
+  @Override
+  public void close() throws TsdbException {
+    try {
+      tableSession.close();
+    } catch (IoTDBConnectionException ioTDBConnectionException) {
+      LOGGER.error("Failed to close TableSession");
+      throw new TsdbException(ioTDBConnectionException);
+    }
+  }
+
+  @Override
+  public void createSchemaTemplate(Template template) {
+    // nothing to do
```

Comment:
Would throwing an UnsupportedOperationException be better?

Reply:
+1， it will expose our bugs early